### PR TITLE
[CN-687] Custom Logger support for the go client

### PR DESCRIPTION
**Pull request by semihbkgr**
## Description

Implementing Hazelcast client custom logger adapter for `logr` library which is used in the operator.

For now, it is only for debug-purpose. So there is no config field to set the level of Hz client logger. The level is set to `logger.InfoLevel` in hard-coded way.

### CODE REVIEW

File: internal/hazelcast-client/logger.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+type LogrHzClientLoggerAdapter struct {
+	logger     logr.Logger
+	weight     logger.Weight
+	enableFunc func() bool
+}
+
+func NewLogrHzClientLoggerAdapter(rootLogger logr.Logger, level logger.Level, h *hazelcastv1alpha1.Hazelcast) (*LogrHzClientLoggerAdapter, error) {
+	w, err := logger.WeightForLogLevel(level)
+	l := rootLogger.WithName("client").WithName(h.Name).
```
Comment (mtyazici) [inline internal/hazelcast-client/logger.go:18]: rootLogger's value might differ when client is created from different reconcilers, do we have a way of reverting that value?

Comment (semihbkgr) [reply internal/hazelcast-client/logger.go:18]: `rootLogger` does not have any value in the current use case. It is from `ctrl.Log` controller-runtime logger. In the future, we may want to pass more specific root logger with its values.

Comment (mtyazici) [inline internal/hazelcast-client/logger.go:18]: We use `h.Name` two times, should drop one of them.

**Review by mtyazici — APPROVED** [review]
LGTM :+1:

File: internal/hazelcast-client/logger.go (hunk 1 above)
Comment (semihbkgr) [reply internal/hazelcast-client/logger.go:18]: `.WithName(h.Name)` to make client logger name unique
`WithValues("Hazelcast", h.Name)` to enable filtering HZ CR logs with their client logs

**Review by hasancelik — APPROVED** [review]

**Pull request merged**